src/OnboardingSPA/components/NewfoldInterfaceSkeleton/SiteGen/index.js

```diff
@@ -172,25 +181,69 @@ const SiteGen = () => {
 			}
 
 			// If the retry also did not work show the error state
-			setFailedApi( ( prevState ) => {
+			setFailedSiteMetaAPIs( ( prevState ) => {
 				// If the error doesn't exist add it to the Failed List
 				if ( ! prevState.includes( identifier ) ) {
 					return [ ...prevState, identifier ];
 				}
+
 				return prevState;
 			} );
-			// Activate the Error Page
-			currentData.sitegen.siteGenErrorStatus = true;
-			setCurrentOnboardingData( currentData );
+
+			if ( siteGenErrorStatus === false ) {
+				updateSiteGenErrorStatus( true );
+				setIsGeneratingSiteMeta( false );
+			}
+
+			if ( window.nfdOnboarding.siteGenTimerInterval ) {
+				clearInterval( window.nfdOnboarding.siteGenTimerInterval );
+			}
+
+			return;
 		}
-	}
 
-	async function generateSiteGenData( forceRun = false ) {
-		// ForceRun tells us to bypass the page check or not
-		// because the requests failed on Logo step is retried in Experience step
+		// Sets the Site Title and Taglin in Live Preview
+		if ( identifier === 'site_config' ) {
+			editEntityRecord( 'root', 'site', undefined, {
+				title: data.body.site_title,
+				description: data.body.tagline,
+			} );
+		}
 
+		// A Identifier request was successfully made with valid response
+		currentData.sitegen.siteGenMetaStatus.currentStatus += 1;
+
+		if (
+			currentData.sitegen.siteGenMetaStatus.currentStatus ===
+			currentData.sitegen.siteGenMetaStatus.totalCount
+		) {
+			// Once all requests are completed use cache to get data
+			currentData.sitegen.skipCache = false;
+			// Increase count after site meta calls to ensure systematic call of homepages
+			currentData.sitegen.siteGenMetaStatus.totalCount += 1;
+			// Get the homepages and set that in flow
+			const response = await getHomepages(
+				currentData.sitegen.siteDetails.prompt
+			);
+
+			if ( response.body ) {
+				currentData.sitegen.homepages.data = response.body;
+				currentData.sitegen.siteGenMetaStatus.currentStatus += 1;
```

Comment:
<img width="2046" alt="Screenshot 2024-04-05 at 11 10 23 PM" src="https://github.com/newfold-labs/wp-module-onboarding/assets/4736718/6b9e7e01-e162-47cd-89e2-34ae421e3f4f">
<img width="2045" alt="Screenshot 2024-04-05 at 11 24 52 PM" src="https://github.com/newfold-labs/wp-module-onboarding/assets/4736718/d51478d0-f166-4840-a283-f648bf2ba4b3">

i failed the homepage API with mock error. 
When the home page API fails in the background on the experience step and the user has selected one of the options from the survey question. The page doesn't move to the next step and the user is not aware of any errors, the user will be stuck on that step.